Analytics/Classes/Internal/SEGUtils.m

```diff
@@ -34,4 +34,62 @@ + (id _Nullable)plistFromData:(NSData *_Nonnull)data
     return plist;
 }
 
+
++(id)traverseJSON:(id)object andReplaceWithFilters:(NSDictionary<NSString*, NSString*>*)patterns
+{
+    if (object == nil || object == NSNull.null || [object isKindOfClass:NSNull.class]) {
+        return object;
+    }
+    
+    if ([object isKindOfClass:NSDictionary.class]) {
+        NSDictionary* dict = object;
+        NSMutableDictionary* newDict = [NSMutableDictionary dictionaryWithCapacity:dict.count];
+        
+        for (NSString* key in dict.allKeys) {
+            newDict[key] = [self traverseJSON:dict[key] andReplaceWithFilters:patterns];
+        }
+        
+        return newDict;
+    }
+    
+    if ([object isKindOfClass:NSArray.class]) {
+        NSArray* array = object;
+        NSMutableArray* newArray = [NSMutableArray arrayWithCapacity:array.count];
+        
+        for (int i = 0; i < array.count; i++) {
+            newArray[i] = [self traverseJSON:array[i] andReplaceWithFilters:patterns];
+        }
+        
+        return newArray;
+    }
+
+    if ([object isKindOfClass:NSString.class]) {
+        NSError* error = nil;
+        NSMutableString* str = [object mutableCopy];
+        
+        for (NSString* pattern in patterns) {
+            NSRegularExpression* re = [NSRegularExpression regularExpressionWithPattern:pattern
+                                                                                options:0
+                                                                                  error:&error];
+            
+            if (error) {
+                @throw error;
+            }
+            
+            NSInteger matches = [re replaceMatchesInString:str
+                                                   options:0
+                                                     range:NSMakeRange(0, str.length)
+                                              withTemplate:patterns[pattern]];
+            
+            if (matches > 0) {
+                SEGLog(@"%@ Redacted value from action: %@", self, str);
+                
+                return str;
```

Comment:
It looks like we're trying to match multiple patterns, but here we return immediately after the first match? Should this return statement be outside the `for` loop?

Reply:
Good catch! It's fixed